tests/foreman/longrun/test_provisioning_computeresource.py

```diff
@@ -23,88 +22,101 @@
-        cls.rhev_password = settings.rhev.password
-        cls.rhev_username = settings.rhev.username
-        cls.rhev_datacenter = settings.rhev.datacenter
-        cls.rhev_img_name = settings.rhev.image_name
-        cls.rhev_img_arch = settings.rhev.image_arch
-        cls.rhev_img_os = settings.rhev.image_os
-        cls.rhev_img_user = settings.rhev.image_username
-        cls.rhev_img_pass = settings.rhev.image_password
-        cls.rhev_vm_name = settings.rhev.vm_name
-        cls.rhev_storage_domain = settings.rhev.storage_domain
-        cls.rhv_api = RHEVMSystem(
-            hostname=cls.rhev_url.split('/')[2],
-            username=cls.rhev_username,
-            password=cls.rhev_password,
-            version='4.0',
-            verify=False,
-        )
-        cls.cluster_id = cls.rhv_api.get_cluster(cls.rhev_datacenter).id
-        cls.storage_id = cls.rhv_api.get_storage_domain(cls.rhev_storage_domain).id
-        cls.network_id = (
-            cls.rhv_api.api.system_service().networks_service().list(search=f'name={bridge}')[0].id
-        )
-        if is_open('BZ:1685949'):
-            dc = cls.rhv_api._data_centers_service.list(search=f'name={cls.rhev_datacenter}')[0]
-            dc = cls.rhv_api._data_centers_service.data_center_service(dc.id)
-            cls.quota = dc.quotas_service().list()[0].id
-        else:
-            cls.quota = 'Default'
-
-        # Vmware Settings
-        cls.vmware_server = settings.vmware.vcenter
-        cls.vmware_password = settings.vmware.password
-        cls.vmware_username = settings.vmware.username
-        cls.vmware_datacenter = settings.vmware.datacenter
-        cls.vmware_img_name = settings.vmware.image_name
-        cls.vmware_img_arch = settings.vmware.image_arch
-        cls.vmware_img_os = settings.vmware.image_os
-        cls.vmware_img_user = settings.vmware.image_username
-        cls.vmware_img_pass = settings.vmware.image_password
-        cls.vmware_vm_name = settings.vmware.vm_name
-        cls.current_interface = VMWARE_CONSTANTS.get('network_interfaces') % bridge
-        cls.vmware_api = VMWareSystem(
-            hostname=cls.vmware_server, username=cls.vmware_username, password=cls.vmware_password
-        )
-        cls.vmware_net_id = cls.vmware_api.get_network(cls.current_interface)._moId
-
-        # Provisioning setup
-        cls.org = entities.Organization(name=gen_string('alpha')).create()
-        cls.org_name = cls.org.name
-        cls.loc = entities.Location(name=gen_string('alpha'), organization=[cls.org]).create()
-        cls.loc_name = cls.loc.name
-        cls.config_env = configure_provisioning(
-            compute=True, org=cls.org, loc=cls.loc, os=cls.rhev_img_os
-        )
-        cls.os_name = cls.config_env['os']
 
-    def tearDown(self):
-        """Delete the host to free the resources"""
-        super().tearDown()
-        hosts = Host.list({'organization': self.org_name})
+@pytest.fixture(scope="module")
+def rhev(bridge):
```

Comment:
While at this time its a fair amount of syntax refactoring, I think the original test had a better pattern with attributized reference to these key/value pairs.

You can do this very simply by defining a `collections.namedtuple` type, and use it to store data from `settings`.

Its not clear to me why all of these key/value pairs have to be 'translated', and stored with `rhev_<key>`, when the test appears to be building dictionaries that reference the original `<key>` without any prefix.  It appears the test could directly reference `settings.rhev`, and these fixtures would be far less verbose.